src/backend/catalog-builder/catalog-builder.lambda.ts

```
@@ -0,0 +1,253 @@
+import { env } from 'process';
+import { gunzip } from 'zlib';
+
+import type { AssemblyTargets } from '@jsii/spec';
+// eslint-disable-next-line import/no-unresolved
+import type { Context } from 'aws-lambda';
+import { AWSError, S3 } from 'aws-sdk';
+import { SemVer } from 'semver';
+import { extract } from 'tar-stream';
+
+let s3: S3 | undefined;
+
+export const CATALOG_OBJECT_KEY = 'catalog.json';
+
+const KEY_FORMAT_REGEX = /^packages\/((?:@[^/]+\/)?[^/]+)\/v([^/]+)\/.*$/;
+// Capture groups:                   ┗━━━━━━━━1━━━━━━━━━┛   ┗━━2━━┛
+
+/**
+ * Regenerates the `catalog.json` object in the configured S3 bucket.
+ *
+ * @param event configuration for the rebuild job. In particular, the `rebuild`
+ *              property can be set to `true` in order to trigger a full (i.e:
+ *              non-incremental) rebuild of the object.
+ * @param context the lambda context in which this execution runs.
+ *
+ * @returns the information about the updated S3 object.
+ */
+export async function handler(event: { readonly rebuild?: boolean }, context: Context) {
```

What happens if for any reason an exception is raised? Usually it is advised to add try & catch for the entire lambda code. There was a bug two years ago (and I don't know if it's fixed already) that when exceptions were not handled on the lambda function code, you couldn't see any information about it on cloudwatch. There was another bug where it could stale your Dynamodb trigger. So.. generally saying we should probably use try & catch for every lambda function and be intentional about the way we want to handle exceptions

---

So this particular handler is `async`, so it returns a `Promise` that'll `reject` in case an exception is thrown. This will cause the Lambda to fail, and the failure details of the execution will be the error that was "thrown" here.

So it may not show in CloudWatch Logs, but it'll actually show up where it's relevant (messages in the DLQ have an attribute that provides the Lambda failure cause).